evennia/prototypes/protfuncs.py

```diff
@@ -325,3 +326,15 @@ def objlist(*args, **kwargs):
 
     """
     return ["#{}".format(obj.id) for obj in _obj_search(return_list=True, *args, **kwargs)]
+
+
+def dbref(*args, **kwargs):
+    """
+    Usage $dbref(<#dbref>)
+    Returns one Object searched globally by #dbref. Error if #dbref is invalid.
+    """
+    _RE_DBREF = re.compile(r"\#[0-9]+")
+    if not args or len(args) < 1 or _RE_DBREF.match(args[0]) is None:
+        raise ValueError('$dbref requires a valid #dbref argument.')
```

Comment:
This is correct - for a `protfunc` (as opposed to their sibling, the `inlinefunc` which uses the same parsing engine), it's okay to raise an exception - this will abort the spawning and allow the user to correct the issue.

Reply:
👍